packages/compiler/src/core/linter.ts

```diff
@@ -53,13 +60,18 @@ export function resolveLinterDefinition(
 
 export function createLinter(
   program: Program,
-  loadLibrary: (name: string) => Promise<LibraryInstance | undefined>,
+  nameResolver: NameResolver,
```

Comment:
this is not really how I would have seens that, with this we basically make the linter engine tied to those built in rules which are not really the point. This would break any programmatic usage of the linter.

Could you instead maybe add a function that allows to register some rule on the fly and keep the logic for the builtin linter in its dedicated area

Reply:
updated. thanks.